modules/data.land/R/soilgrids_soc_extraction.R

```diff
@@ -0,0 +1,199 @@
+  lonlat <- cbind(internal_site_info$site_info.lon, internal_site_info$site_info.lat)
+  base_data_url <- "/vsicurl?max_retry=30&retry_delay=60&list_dir=no&url=https://files.isric.org/soilgrids/latest/data/ocd/ocd_"
+  depths <- c("0-5cm", "5-15cm", "15-30cm", "30-60cm", "60-100cm", "100-200cm")
+  layer_thick <- c(0.05,0.10,0.15,0.30,0.40,1.00) # in unit m
+ 
+  # reproject locations to soilgrids projection
+  #Soilgrids data is using Homolosine projection https://www.isric.org/explore/soilgrids/faq-soilgrids 
+  p <- terra::vect(lonlat, crs = "+proj=longlat +datum=WGS84") # Users need to provide lon/lat
+  newcrs <- "+proj=igh +datum=WGS84 +no_defs +towgs84=0,0,0" 
+  p_reproj <- terra::project(p, newcrs) # Transform the point vector to data with Homolosine projection
+  
+  # setup progress bar
+  if (verbose) {
+    j <- 1
+    pb <- utils::txtProgressBar(min = 0, max = length(depths), char="*", width=70, style = 3)
+  }
+  
+  for (dep in seq_along(depths)) {
+    # setup virtual raster URLs for each layer
+    ocd_mean.url <- paste0(base_data_url,depths[dep],"_mean.vrt")
+    ocd_Q0.05.url <- paste0(base_data_url, depths[dep], "_Q0.05.vrt")
+    ocd_Q0.50.url <- paste0(base_data_url, depths[dep], "_Q0.5.vrt")
+    ocd_Q0.95.url <- paste0(base_data_url, depths[dep], "_Q0.95.vrt")
+    
+    # create virtual rasters && extract SOC values - the original unit is hg/m3
+    ocd_mean <- terra::extract(terra::rast(ocd_mean.url), p_reproj)
+    ocd_Q0.05_map <- terra::extract(terra::rast(ocd_Q0.05.url), p_reproj)
+    ocd_Q0.50_map <- terra::extract(terra::rast(ocd_Q0.50.url), p_reproj)
+    ocd_Q0.95_map <- terra::extract(terra::rast(ocd_Q0.95.url), p_reproj)
+    
+    #change the unit to more common kg/m3
+    ocd_mean_real <- ocd_mean[, -1] / 10
+    ocd_Q0.05_real <- ocd_Q0.05_map[, -1] / 10
+    ocd_Q0.50_real <- ocd_Q0.50_map[, -1] / 10
+    ocd_Q0.95_real <- ocd_Q0.95_map[, -1] / 10
+    
+    ocdquant[dep, ] <-c(ocd_mean_real,ocd_Q0.05_real,ocd_Q0.50_real,ocd_Q0.95_real)
+    ### Display progress to console
+    if (verbose) {
+      utils::setTxtProgressBar(pb, j)
+      j <- j+1
+      utils::flush.console()}
+  
+  # cleanup interim results
+    rm(ocd_mean.url, ocd_Q0.05.url, ocd_Q0.50.url, ocd_Q0.95.url, 
+       ocd_mean, ocd_Q0.05_map, ocd_Q0.50_map, ocd_Q0.95_map,
+       ocd_mean_real, ocd_Q0.05_real, ocd_Q0.50_real, ocd_Q0.95_real)
+  }
+  
+
+  if (verbose) {
+    close(pb)
+  }
+  
+ # parse extracted data and prepare for output
+  quantile_name <-c(paste("Mean_",site_info$id,sep=""),paste("0.05_",site_info$id,sep=""),paste("0.5_",site_info$id,sep=""),paste("0.95_",site_info$id,sep=""))
+  colnames(ocdquant) <- quantile_name
+  ocdquant_dep <- cbind(ocdquant,depths)
+  ocdquant_long <- tidyr::pivot_longer(as.data.frame(ocdquant_dep),cols=tidyselect::all_of(quantile_name))
+  ocd_df <- tidyr::separate(ocdquant_long,name, c('Quantile', 'siteid'),"_")
```

Comment:
@Qianyuxuan @serbinsh this is where the undefined `name` is coming from.

Reply:
"name" is a default column name after using "pivot_longer". I guess it should be fixed with an assigned name. I will try it.

Reply:
@Qianyuxuan the issue isn't that the name is default, it's that because it's a column in the dataframe it's not defined outside of the dataframe. The solution is to add
```
##' @importFrom rlang .data
```
to the Roxygen for this function and then, in that line, to refer to the variable as `.data$name` instead of just `name`. Here .data is a reference to the dataframe being passed through the pipe. Normally this is implicit, but R's code checking forces us to make this explicit. This is one of rare cases where it actually does make more sense to do the importFrom at the function level, rather than using namespaces.

Reply:
@mdietze Just saw your comments. But I used another way to avoid the issue. You can test it to see if it is fixed.